cluster/discovery/discovery.go

```diff
@@ -0,0 +1,33 @@
+package discovery
+
+// NodeEntry is used to discovery nodes in the cluster
+type NodeEntry struct {
+	Id            string
+	Ip            string
+	GossipVersion string
+}
+
+// ClusterInfo is the cluster info used while discoveryping nodes
+// and discovering peer nodes using gossip
+type ClusterInfo struct {
+	Size    int
+	Nodes   map[string]NodeEntry
+	Version uint64
```

Comment:
In case of a kvdb discovery service it will be KvdbVersion. I will add a comment.